src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -107,8 +108,16 @@ public async Task RunAsync()
             }
             else
             {
-                file = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), "node", "bin", $"node{IOUtil.ExeExtension}");
+                bool useNode10 = ExecutionContext.Variables.GetBoolean("AGENT_USE_NODE10") ??
+                    StringUtil.ConvertToBoolean(System.Environment.GetEnvironmentVariable("AGENT_USE_NODE10"), false);
+
+                string nodeFolder = (Data is Node10HandlerData || useNode10) ? "node10" : "node";
+
+                Trace.Info($"Task.json has node10 handler data: {Data is Node10HandlerData}, use node10 for node tasks: {useNode10}");
```

Comment:
you have `Data is Node10HandlerData` twice, maybe store the bool result to some local variable.